packages/tupaia-web/src/features/EntitySearch/SearchBar.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { TextField, TextFieldProps } from '@material-ui/core';
+import SearchIcon from '@material-ui/icons/Search';
+import styled from 'styled-components';
+
+const SearchInput = styled(TextField).attrs({
+  variant: 'outlined',
+  placeholder: 'Search location',
+  fullWidth: true,
+  InputProps: {
+    startAdornment: <SearchIcon />,
+  },
+})<TextFieldProps>`
+  .MuiInputBase-root {
+    background: ${({ theme }) => theme.searchBar.background};
+    border-radius: 2.7rem;
+  }
+
+  .MuiOutlinedInput-notchedOutline {
+    border: 2px transparent solid;
+  }
+
+  &:hover .MuiOutlinedInput-notchedOutline {
+    border: 2px ${({ theme }) => theme.form.border} solid;
+  }
+
+  .Mui-focused .MuiOutlinedInput-notchedOutline {
+    border: 2px ${({ theme }) => theme.palette.primary.main} solid;
+  }
+
+  .MuiInputBase-input {
+    padding: 0.6em;
+  }
+`;
+
+interface SearchBarProps {
+  value?: string;
+  onChange: (newValue: string) => void;
+  onFocusChange: (isFocused: boolean) => void;
+}
+
+export const SearchBar = ({ value = '', onChange, onFocusChange }: SearchBarProps) => {
+  const onChangeInputValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.value);
+  };
+
+  return (
+    <SearchInput value={value} onChange={onChangeInputValue} onFocus={() => onFocusChange(true)} />
```

Comment:
Not sure, but would it be useful to have an `onBlur` event listener here too?

Reply:
I initially had the onBlur event closing the entity menu but then I realised that the menu kept closing whenever I clicked on it and realised the blur was the problem so I don't think we want the blur for that interaction. Is there any thing else that you think should happen when you blur?